eng/pipelines/templates/steps/build-test.yml

```diff
@@ -9,28 +15,31 @@ steps:
     displayName: 'Download Tools Archive From Github Release and Extract it'
 
   - task: UsePythonVersion@0
-    displayName: 'Use Python $(PythonVersion)'
+    displayName: 'Use Python ${{ parameters.PythonVersion }}'
     inputs:
-      versionSpec: '$(PythonVersion)'
+      versionSpec: '${{ parameters.PythonVersion }}'
 
   - task: PythonScript@0
     displayName: 'Run VerifyAgentOS script'
     inputs:
       scriptPath: '$(Build.BinariesDirectory)/sdk-tools/scripts/python/verify_agent_os.py'
-      arguments: $(OSName)
+      arguments: ${{ parameters.OSName }}
 
   - script: |
       pip install pathlib twine codecov beautifulsoup4
     displayName: 'Prep Environment'
 
+  - ${{ parameters.BeforeTestSteps }}
```

Comment:
It doesn't look like this is used by anyone. Do you have plans for people to use it? If not I would remove it for now.

Reply:
The plan is to use it for KeyVault. Individual tests for KeyVault can be run in parallel for considerable time savings. 

In the case of KeyVault we will use this step to pip install a package that enables us to run the parallel tests. That way we keep the parallel test work scoped just to the services that support it. If it works well we use this method in other places we may push the package up into the `Prep Environment` step and remove `BeforeTestSteps`.